xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/skipnav.vm

```diff
@@ -0,0 +1,34 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+## Provide a way to skip directly to the main content of page
+## See: https://www.w3.org/TR/WCAG20-TECHS/G1.html
+<div class="skip-nav">
+  #if ("$!xcontext.action" == 'view' || "$!xcontext.action" == 'preview')
+    ## Point at #xwikicontent to skip over the navigation tools including the breadrumb
+    #set ($targetNav = '#xwikicontent')
+  #elseif (("$!xcontext.action" == 'inline' || "$!xcontext.action" == 'edit') && !$editConfirmation)
+    ## In edit mode, point at the title field
+    #set ($targetNav = '#editMeta')
+  #else
+    ## Point at '#mainContentArea' which is present in all pages that include "startpage.vm"
+    #set ($targetNav = '#mainContentArea')
+  #end
```

Comment:
I wonder if it wouldn't make sense to make this consistent with the main page region as this if/else seems to be quite fragile.

Reply:
`targetNav` is used as an href. I'm pretty sure we need the **id** of an element to have a proper href.
The id of the main region as introduced in https://github.com/xwiki/xwiki-platform/pull/2166 is **not defined** (no value, or value sensitive to the template). This could be a change I need to make on PR 2166 before we merge it.